benchmarks/triangulated_surface_distance/generator.hpp

```diff
@@ -0,0 +1,252 @@
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX_HyperTriangle.hpp>
+
+#include <Kokkos_Array.hpp>
+#include <Kokkos_MathematicalConstants.hpp>
+
+static auto icosahedron()
+{
+  auto a = Kokkos::numbers::phi_v<float>;
+  auto b = 1.f;
+
+  using Point = ArborX::ExperimentalHyperGeometry::Point<3>;
+
+  std::vector<Point> vertices;
+
+  vertices.push_back(Point{0, b, -a});
+  vertices.push_back(Point{b, a, 0});
+  vertices.push_back(Point{-b, a, 0});
+  vertices.push_back(Point{0, b, a});
+  vertices.push_back(Point{0, -b, a});
+  vertices.push_back(Point{-a, 0, b});
+  vertices.push_back(Point{0, -b, -a});
+  vertices.push_back(Point{a, 0, -b});
+  vertices.push_back(Point{a, 0, b});
+  vertices.push_back(Point{-a, 0, -b});
+  vertices.push_back(Point{b, -a, 0});
+  vertices.push_back(Point{-b, -a, 0});
+
+  std::vector<Kokkos::Array<int, 3>> triangles;
+
+  triangles.push_back({2, 1, 0});
+  triangles.push_back({1, 2, 3});
+  triangles.push_back({5, 4, 3});
+  triangles.push_back({4, 8, 3});
+  triangles.push_back({7, 6, 0});
+  triangles.push_back({6, 9, 0});
+  triangles.push_back({11, 10, 4});
+  triangles.push_back({10, 11, 6});
+  triangles.push_back({9, 5, 2});
+  triangles.push_back({5, 9, 11});
+  triangles.push_back({8, 7, 1});
+  triangles.push_back({7, 8, 10});
+  triangles.push_back({2, 5, 3});
+  triangles.push_back({8, 1, 3});
+  triangles.push_back({9, 2, 0});
+  triangles.push_back({1, 7, 0});
+  triangles.push_back({11, 9, 6});
+  triangles.push_back({7, 10, 6});
+  triangles.push_back({5, 11, 4});
+  triangles.push_back({10, 8, 4});
+
+  return std::make_tuple(vertices, triangles);
+}
+
+void convertTriangles2EdgeForm(std::vector<Kokkos::Array<int, 3>> &triangles,
+                               std::vector<Kokkos::Array<int, 2>> &edges)
+{
+  std::map<std::pair<int, int>, int> hash;
+
+  edges.clear();
+  for (int i = 0, eindex = 0; i < (int)triangles.size(); ++i)
+  {
+    int e[3];
+    for (int j = 0; j < 3; ++j)
+    {
+      auto minmax_pair =
+          std::minmax(triangles[i][j], triangles[i][(j + 1) % 3]);
+      auto it = hash.find(minmax_pair);
+      if (it == hash.end())
+      {
+        edges.push_back({minmax_pair.first, minmax_pair.second});
+        hash[minmax_pair] = eindex;
+        e[j] = eindex++;
+      }
+      else
+      {
+        e[j] = it->second;
+      }
+    }
+    triangles[i] = {e[0], e[1], e[2]};
+  }
+}
+
+template <typename ExecutionSpace, typename MemorySpace>
+void convertTriangles2VertexForm(
+    ExecutionSpace const &space,
+    Kokkos::View<Kokkos::Array<int, 3> *, MemorySpace> &triangles,
```

Comment:
Did we ever agree on how to pass output views when no allocation occurs.  I am not a big fan of using non-const references to denote that a view is being written into.  Non-const can also be interpreted as it can be resized/realloc'd or bound to some other view.

Reply:
> Did we ever agree on how to pass output views when no allocation occurs.

No, we don't decided on a guideline.

> I am not a big fan of using non-const references to denote that a view is being written into.

I prefer it this way, but open to arguments.

> Non-const can also be interpreted as it can be resized/realloc'd or bound to some other view.

Indeed it can.